### PR TITLE
Added User Session With Expiry Window

### PR DESCRIPTION
**Pull request by zenovy**

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ class StageConfig(Config):
...
     TESTING = True
 
-
 class DevelopmentConfig(Config):
     DEBUG = True
     ENV = 'development'
-    TESTING = True
\ No newline at end of file
+    TESTING = True
+    SECRET_KEY = 'default_secret'
```
Comment (naomiajacobs) [inline config.py]: Where is this used?

File: app/session_util.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from flask import session
+
+import time
```
[the remark this answers is not in the document]
Comment (zenovy) [reply app/session_util.py]: I'm fine with the ordering but don't really want to commit to doing that manually, maybe in the future we can add a linter which does that?

File: app/__init__.py [hunk 3]
```diff
@@ -12,9 +13,14 @@
 from app import models
 from app import routes
 
+@app.before_request
+def make_session_permanent():
+    session.permanent = True
+    app.permanent_session_lifetime = timedelta(minutes=5)
```
[the remark this answers is not in the document]
Comment (zenovy) [reply app/__init__.py]: Yeah, that was a miss on my part.

File: app/controllers.py [hunk 4]
```diff
@@ -2,15 +2,18 @@
...
+from app.session_util import is_user_logged_in, create_session, delete_session
 
 class HomeController():
     def handle(self):
-        user = User.query.first()
-        if user:
-            return f"Hello, {user.email}!"
-        else:
-            return "No users. Sad!"
+        if not is_user_logged_in():
```
[the remark this answers is not in the document]
Comment (zenovy) [reply app/controllers.py]: Agreed!

File: config.py (hunk 1 above)
Comment (zenovy) [reply config.py]: It's used by Flask's session feature to secure the cookie!

File: app/session_util.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+expiry_window = 604800 # 7 days
+
+def is_user_logged_in():
+    expiryTime = session.get('sessionExpiryTimestamp', None)
+    if expiryTime == None or is_session_expired():
```
[the remark this answers is not in the document]
Comment (zenovy) [reply app/session_util.py]: No need, Flask handles expiration. Forgot to remove the manual expiration code. Fixed.

**Pull request merged**